doc/spec_dataValidityPropagation.dox

```diff
@@ -0,0 +1,90 @@
+
+### 2.1 MetaDataPropagatingRegisterDecorator (*)
+
+- 2.1.1 Each input and each output of a module (or fan out) is decorated with a MetaDataPropagatingRegisterDecorator (except for the TriggerFanOut, see. \ref dataValidity_2_4 "2.4")
+- 2.1.2 The decorator knows about the module it is connected to. It is called the 'owner'.
+
+- 2.1.3 **read:** For each read operation it checks the incoming data validity and increases/decreases the data fault counter of the owner.
+- 2.1.5 **write:** When writing, the decorator is checking the validity of the owner and the individual flag of the output set by the user. Only if both are 'ok' the output validity is 'ok', otherwise the outgoing data is send as 'faulty'.
+
+### 2.2 removed
+
+### 2.3 ApplicationModule
+
+- 2.3.1 Each ApplicationModule has one data fault counter variable which is increased/decreased by EntityOwner::incrementDataFaultCounter() and  EntityOwner::decrementDataFaultCounter.
+- 2.3.2 All inputs and outputs have a MetaDataPropagatingRegisterDecorator.
+- \anchor dataValidity_2_3_3 2.3.3 The main loop of the module usualy does not care about data validity. If any input is invalid, all outputs are automatically invalid \ref dataValidity_comment_2_3_3a "(*)". The loop just runs through normaly, even if an input has invalid data. \ref dataValidity_comment_2_3_3b "(*)"
+- 2.3.4 Inside the ApplicationModule main loop the module's data fault counter is accessible. The user can increment and decrement it, but has to be careful to do this in pairs. The more common use case will be to query the module's data validity.
+
+### 2.4 TriggerFanOut
+\anchor dataValidity_2_4
+ 
+The TriggerFanOut is special in the sense that it does not compute anything, but reads multiple independent poll-type inputs when a trigger arrives, and pushes them out. In contrast to an ApplicationModule or one of the data fan-outs, their data validities are not connected.
+
+- 2.4.1 Only the push-type trigger input of the TriggerFanOut is equiped with a MetaDataPropagatingRegisterDecorator.
+- \anchor dataValidity_2_4_2 2.4.2 The poll-type data inputs do not have a MetaDataPropagatingRegisterDecorator. \ref dataValidity_comment_2_4_2 "(*)".
+
+- 2.4.3 The individual poll-type inputs propagate the data validity flag only to the corresponding outputs.
+- \anchor dataValidity_2_4_4 2.4.4 Although the trigger conceptually has data type 'void', it can also be `faulty` \ref dataValidity_comment_2_4_4 "(*)". An invalid trigger will result be processed, but all data that is being read out is flagged as `faulty`.
+
+### 2.5 Interaction with exception handling
+
+See @ref spec_execptionHandling.
+
+- 2.5.1 When creating the objects, ApplicationCore takes care that the MetaDataPropagatingRegisterDecorator is always placed around the ExceptionHandlingDecorator if both decorators are used. Like this a `faulty` flag raised by the ExceptionHandlingDecorator is automatically picked up by the MetaDataPropagatingRegisterDecorator.
```

Comment:
```suggestion
- 2.5.1 When creating objects, ApplicationCore takes care that the MetaDataPropagatingRegisterDecorator is always placed around the ExceptionHandlingDecorator if both decorators are used. Like this a `faulty` flag raised by the ExceptionHandlingDecorator is automatically picked up by the MetaDataPropagatingRegisterDecorator.
```

Reply:
You proposal is not correct. It reads as if there always is a MetaDataPropagatingRegisterDecorator around an ExceptionHandlingDecorator, which is not the case. Most accessors only have MetaDataPropagatingRegisterDecorator, a few only have ExceptionHandlingDecorator and some have both. If both are placed, the order is important.
How about this.
```
The MetaDataPropagatingRegisterDecorator is always placed around the ExceptionHandlingDecorator if both decorators are used on a process variable.
```

Reply:
How about:
```
When used together (on a Process Variable) the MetaDataPropagatingRegisterDecorator always decorates the ExceptionHandlingDecorator
```